src/Block/Fetch.hs

```diff
@@ -66,67 +72,99 @@ data OgmiosInfo = OgmiosInfo
 
 newtype OgmiosWorkerMVar = MkOgmiosWorkerMVar (MVar (Async ()))
 
+newtype ControlApiToken = ControlApiToken (Maybe String)
+  deriving stock (Eq, Show)
+
+withControlApiTokenToken ::
+  (MonadReader r m, Has ControlApiToken r) =>
+  e ->
+  Maybe String ->
+  m (Either e ()) ->
+  m (Either e ())
+withControlApiTokenToken err token action = do
+  ControlApiToken expectToken <- ask
+  case () of
+    _
+      | isNothing expectToken -> action
```

Comment:
If there's no specified expected token, action shouldn't be performed. Secure by default

Reply:
Any reason not to use MultiwayIf here instead?